### DISCUSSION TITLE
Pseudo Residuals for multivariate normal emission distribution.

### DISCUSSION THREAD

**Discussion by threex1ng** [bmcclintock/momentuHMM#71] [Q&A] [answered]
Hi Brett,

I am trying to compute pseudo residuals for HMM assuming bivariate Gaussian as emission distribution. I noticed that you made use of the Mahalanobis distance to compute the cumulative density function for bivariate Gaussian (mvnorm2). So that the cumulative probability and pseudo residual can be calculated for each time t. 
There are two kinds of pseudo-residual, one is uniform pseudo-residual and the other is normal pseudo-residual according to Zucchini et al book.

However, in your code for pseudoRes function:
if(dist[[j]] %in% mvndists){
          genRes[[paste0(j,"Res")]][i] <- stats::qchisq(genRes[[paste0(j,"Res")]][i],df=ndim)
        }
it seems that you assumed the pseudo-residuals are distributed Chi-squared with ndim degree of freedom instead of Normal.
May I know the reason why you choose qchisq?
Thank you for your useful package and look forward to your reply.

Bests,
Jiaxing

**Comment by bmcclintock** [comment]
Hi Jiaxing,

Good question! The short answer is that the squared Mahalanobis distance of a multivariate Gaussian distribution is Chi-Square distributed. Markus Thill has written a very nice [post](https://markusthill.github.io/mahalanbis-chi-squared/) on this.

Cheers,
Brett

**Comment by bmcclintock** [comment, marked as answer]
On second thought, I didn't think thoroughly enough about your question. Apologies for that. Upon further inspection, I agree that using `stats::qchisq` in the code above is incorrect for calculating the forecast pseudo-residuals for a multivariate normal data stream. This has been changed back to `stats::qnorm` in `momentuHMM` v2.0.0 (fbe6969971010b1cdd92fdda8b9f4d5514c56c02 currently only available in the [develop](/bmcclintock/momentuHMM/tree/develop) branch). Thank you for bringing this to my attention, and apologies again for not catching this bug sooner!
